### PR TITLE
[clang][dataflow] Fix buggy assertion: Compare an unqualified type to an unqualified type.

### PR DESCRIPTION
**Pull request by bazuzi**
Includes crash-reproducing test case.

### CODE REVIEW

**Comment by bazuzi** [comment]
@martinboehme Could you review?

File: clang/unittests/Analysis/FlowSensitive/TransferTest.cpp [hunk 1]
```diff
@@ -3197,6 +3197,26 @@ TEST(TransferTest, AggregateInitialization_NotExplicitlyInitializedField) {
       });
 }
 
+TEST(TransferTest, AggregateInitializationFunctionPointer) {
+  // This is a crash repro.
```
Comment (martinboehme) [inline clang/unittests/Analysis/FlowSensitive/TransferTest.cpp:3201]:
```suggestion
  // This is a repro for an assertion failure.
```

**Review by martinboehme — APPROVED** [review]

**Comment by bazuzi** [comment]
Could you merge for me? I don't have write access.

**Comment by martinboehme** [comment]
> Could you merge for me? I don't have write access.

Done!

**Pull request merged**